File: app/src/main/java/com/savvasdalkitsis/gameframe/injector/infra/network/OkHttpClientInjector.java
package com.savvasdalkitsis.gameframe.injector.infra.network;

import java.util.concurrent.TimeUnit;

import okhttp3.OkHttpClient;
import okhttp3.logging.HttpLoggingInterceptor;

public class OkHttpClientInjector {

    public static OkHttpClient.Builder okHttpClient(int timeout) {
        HttpLoggingInterceptor httpLoggingInterceptor = new HttpLoggingInterceptor();
        httpLoggingInterceptor.setLevel(HttpLoggingInterceptor.Level.BODY);
        return new OkHttpClient.Builder()
                .connectTimeout(timeout, TimeUnit.SECONDS)
                .readTimeout(timeout, TimeUnit.SECONDS)
                .writeTimeout(timeout, TimeUnit.SECONDS)
                .addInterceptor(httpLoggingInterceptor);
    }
}

File: app/src/main/java/com/savvasdalkitsis/gameframe/injector/infra/network/OkHttpClientInjector.kt
package com.savvasdalkitsis.gameframe.injector.infra.network

import java.util.concurrent.TimeUnit

import okhttp3.OkHttpClient
import okhttp3.logging.HttpLoggingInterceptor

object OkHttpClientInjector {

    fun okHttpClient(timeout: Int): OkHttpClient.Builder {
        val httpLoggingInterceptor = HttpLoggingInterceptor()
        httpLoggingInterceptor.level = HttpLoggingInterceptor.Level.BODY
        return OkHttpClient.Builder()
                .connectTimeout(timeout.toLong(), TimeUnit.SECONDS)
                .readTimeout(timeout.toLong(), TimeUnit.SECONDS)
                .writeTimeout(timeout.toLong(), TimeUnit.SECONDS)
                .addInterceptor(httpLoggingInterceptor)
    }
}
